dicomsort

```diff
@@ -40,18 +60,18 @@ while getopts ":rd:s:n" options; do
 done
 
 #check to see if the destdir exists and create it if not
-if [ -e $destdir ]  
-then
+if [ -d $destdir ]; then
 	echo Destination directory exists
 else
 	echo Creating $destdir
 	mkdir $destdir	
 fi
+targdir=$destdir
 
 echo "Procecessing..."
 
-if [ $recurse -eq 1 ]
-then
+#build a list of all the non-hidden files in the target directory
+if [ $recurse -eq 1 ]; then
```

Comment:
in general, I try and wrap my variables in " (e.g. `if [ "$recurse" -eq 1 ];`). If `$recurse` is undefined for whatever reason, the if statement will evaluate to `if [ -eq 1 ];` which will throw a unary operator error. If you use the double quotes it evaluates to `if [ "" -eq 1 ]` which is false

Reply:
Just to make sure I'm really clear, I think all variable access should be wrapped in " regardless of whether or not it's part of an if statement